### DISCUSSION TITLE
Requesting partial responses from GitHub API (selecting output/response fields)

### DISCUSSION THREAD

**Discussion by janakaud** [community/community#21862] [New to GitHub]
<p>Does the GitHub API provide a way of requesting partial responses by selecting/limiting the fields returned in an API call?</p>
<p>APIs like <a>Blogger</a> and <a>Google Drive</a> allow one to specify a set of fields that should be included in the response (e.g. via the <a><code>fields</code> parameter</a>). This is handy for trimming down unnecessary network bandwidth when retrieving large datasets.</p>
<p>Does the GitHub API provide a similar functionality (e.g. retrieving a list of repos that only contains the repo names, excluding all other fields like <code>id</code>, <code>owner</code>, <code>collaborators_url</code>, <code>teams_url</code>, <code>hooks_url</code>, …)? I looked through the <a>documentation</a> but could not find anything useful.</p>
<p>As an example, if I request my own repos using the URL <a href="https://api.github.com/users/janakaud/repos" rel="nofollow noopener">https://api.github.com/users/janakaud/repos</a> I receive a huge response (148 KB deflated):</p>
<pre><code class="lang-auto">[  
{  
"id": 129543465,  
"name": "apps-script-google-services",  
"full\_name": "janakaud/apps-script-google-services",  
"owner": {  
"login": "janakaud",  
"id": 5234574,  
"avatar\_url": "[https://avatars0.githubusercontent.com/u/5234574?v=4](https://avatars0.githubusercontent.com/u/5234574?v=4)",  
...  
},  
"private": false,  
"html\_url": "[https://github.com/janakaud/apps-script-google-services](https://github.com/janakaud/apps-script-google-services)",  
...  
},  
{  
"id": 129543325,  
"name": "apps-script-misc",  
"full\_name": "janakaud/apps-script-misc",  
...  
},  
{  
"id": 129543204,  
"name": "apps-script-monitoring",  
"full\_name": "janakaud/apps-script-monitoring",  
...  
},  
...  
]  
</code></pre>
<p>which contains a whole lot of fields that I am not interested in. I’m expecting to obtain a more compact, filtered response instead, such as:</p>
<pre><code class="lang-auto">[  
{"name": "apps-script-google-services"},  
{"name": "apps-script-misc"},  
{"name": "apps-script-monitoring"},  
...  
]  
</code></pre>
<p>or, even better:</p>
<pre><code class="lang-auto">[  
"apps-script-google-services",  
"apps-script-misc",  
"apps-script-monitoring",  
...  
]  
</code></pre>
<p>TIA :slight_smile:</p>

**Comment by seveas** [comment, marked as answer]
<p>Hi <a class="mention" href="/janakaud">@janakaud</a>,</p>
<p>THe v3 rest api does not provide this, but the v4 GraphQL api is a lot more flexible in this regard.</p>

**Comment by janakaud** [comment]
<p><a class="mention" href="/seveas">@seveas</a> thanks for the insight! Although it’s unfortunate (I have no way of migrating right now); I believe it being available in v4 means that it won’t become available in v3, even in the future</p>
